lib/runtime/wasmer/exports.go

```diff
@@ -110,13 +113,18 @@ func (in *Instance) BabeGenerateKeyOwnershipProof(slot uint64, authorityID [32]b
 		return nil, fmt.Errorf("executing %s: %w", runtime.BabeAPIGenerateKeyOwnershipProof, err)
 	}
 
-	keyOwnershipProof := types.OpaqueKeyOwnershipProof{}
+	var keyOwnershipProof *types.OpaqueKeyOwnershipProof
 	err = scale.Unmarshal(encodedKeyOwnershipProof, &keyOwnershipProof)
 	if err != nil {
 		return nil, fmt.Errorf("scale decoding key ownership proof: %w", err)
 	}
 
-	return keyOwnershipProof, nil
+	// TODO check with team if returning error is preferred or should we keep just returning empty value
+	if keyOwnershipProof == nil {
+		return types.OpaqueKeyOwnershipProof{}, fmt.Errorf("%w", errEmptyKeyOwnershipProof)
+	}
```

Comment:
Saw @kishansagathiya asked this too so answering here. Yes, because the expected type is an option and Gossamer handles options as pointers, where the 0 case is nil